addons/account/models/account_account.py

```diff
@@ -383,9 +383,19 @@ def _search_code(self, operator, value):
         return [('id', 'in', self.with_company(self.env.company.root_id).sudo()._search([('code_store', operator, value)]))]
 
     def _inverse_code(self):
+        code_by_record = {record.id: record.code for record in self}
+
         for record, record_root in zip(self, self.with_company(self.env.company.root_id).sudo()):
             # Need to set record.code with `company = self.env.company`, not `self.env.company.root_id`
-            record_root.code_store = record.code
+            record_root.code_store = code_by_record[record.id]
+
+        # Because changing the code for one company should also change it for all the companies which
+        # share the same root_id, we need to invalidate the cache for all those companies. Because
+        # we can't know which companies share the same root_id unless we do a database search, it's more
+        # efficient to invalidate the cache, which will force recomputation based on `cache_store`.
+        # However, we make sure that we insert back the value for the active company.
```

Comment:
As long as we don't invalidate `code_store`, I don't think we should manipulate the cache directly here. Invalidating is enough (we a way shorter comment about the branches)

Reply:
Thank you so much for the review, that sounds great :+1:

Reply:
(I think we still need to keep the code's value for the current company - all the tests seem to be failing otherwise.)

Reply:
Indeed, wouldn't it make sense that invalidating a computed fields sets it to be recomputed though? 🤔

Reply:
I'll investigate.